tracking-templates/iframe-loader.js

```diff
@@ -37,24 +78,33 @@
                 document.body.appendChild(el);
                 mgr = el;
             };
+            var idtype = '-1';
+            var ccid = '-1';
+            var onid = '-1';
+            var nid = '-1';
+            var name = 'undefined';
+            var isHD = 'undefined';
             var app = mgr.getOwnerApplication(document);
             if (app && app.privateData && app.privateData.currentChannel) {
                 var curr = app.privateData.currentChannel;
-                var idtype = curr.idType || '-1';
-                var ccid = curr.ccid || '-1';
-                var onid = curr.onid || '-1';
-                var nid = curr.nid || '-1';
-                var name = curr.name || 'undefined';
-                var isHD = curr.isHD || 'undefined';
-
-                var req = new XMLHttpRequest();
-                window['{{TRACKING_GLOBAL_OBJECT}}'].getSID(function(sid) {
-                    var m = '?sid=' + sid + '&idtype=' + idtype + '&ccid=' + ccid + '&onid=' + onid + '&nid=' + nid + '&name=' + name + '&isHD=' + isHD;
-                    req.open('GET', '{{SESSION_SERVER_URL}}/meta' + m);
-                    req.send();
-                });
+                idtype = curr.idType || idtype;
+                ccid = curr.ccid || ccid;
+                onid = curr.onid || onid;
+                nid = curr.nid || nid;
+                name = curr.name || name;
+                isHD = curr.isHD || isHD;
             }
-        } catch(e) {}
+            window['{{TRACKING_GLOBAL_OBJECT}}'].getSID(function (sid) {
+                getConsentStatus(function (consentByVendorId) {
+                    var vid = serializeConsentByVendorId(consentByVendorId) || 'undefined';
+                    getSamplerPercentile(function (spc) {
+                        var m = '?sid=' + sid + '&idtype=' + idtype + '&ccid=' + ccid + '&onid=' + onid + '&nid=' + nid + '&name=' + name + '&isHD=' + isHD + '&spc=' + spc + '&vid=' + vid;
+                        var mImg = document.createElement('img');
+                        mImg.setAttribute('src', '{{SESSION_SERVER_URL}}/meta.gif' + m);
```

Comment:
As already mentioned above, I'd only set the keys for which the value is not undefined